2023-09-18 17:41:23.002291: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'cudart64_100.dll'; dlerror: cudart64_100.dll not found
2023-09-18 17:41:23.002457: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
INFO flwr 2023-09-18 17:41:30,639 | app.py:151 | Starting Flower server, config: ServerConfig(num_rounds=10, round_timeout=None)
INFO flwr 2023-09-18 17:41:30,710 | app.py:172 | Flower ECE: gRPC server running (10 rounds), SSL is disabled
INFO flwr 2023-09-18 17:41:30,711 | server.py:86 | Initializing global parameters
INFO flwr 2023-09-18 17:41:30,711 | server.py:273 | Requesting initial parameters from one random client
INFO flwr 2023-09-18 17:41:30,803 | server.py:277 | Received initial parameters from one random client
INFO flwr 2023-09-18 17:41:30,803 | server.py:88 | Evaluating initial parameters
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:41:30,819 | server.py:95 | initial parameters (loss, other metrics): 0.6931471805599453, {'accuracy': 0.46}
INFO flwr 2023-09-18 17:41:30,819 | server.py:101 | FL starting
DEBUG flwr 2023-09-18 17:41:30,937 | server.py:223 | fit_round 1: strategy sampled 2 clients (out of 3)
DEBUG flwr 2023-09-18 17:41:30,969 | server.py:237 | fit_round 1 received 2 results and 0 failures
WARNING flwr 2023-09-18 17:41:30,971 | fedavg.py:243 | No fit_metrics_aggregation_fn provided
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:41:30,978 | server.py:125 | fit progress: (1, 0.4043851898592003, {'accuracy': 0.8066666666666666}, 0.15821749999999923)
DEBUG flwr 2023-09-18 17:41:30,979 | server.py:173 | evaluate_round 1: strategy sampled 3 clients (out of 3)
DEBUG flwr 2023-09-18 17:41:31,009 | server.py:187 | evaluate_round 1 received 3 results and 0 failures
WARNING flwr 2023-09-18 17:41:31,009 | fedavg.py:274 | No evaluate_metrics_aggregation_fn provided
DEBUG flwr 2023-09-18 17:41:31,010 | server.py:223 | fit_round 2: strategy sampled 3 clients (out of 3)
DEBUG flwr 2023-09-18 17:41:31,042 | server.py:237 | fit_round 2 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:41:31,053 | server.py:125 | fit progress: (2, 0.40490823562580686, {'accuracy': 0.8066666666666666}, 0.23332240000000048)
DEBUG flwr 2023-09-18 17:41:31,053 | server.py:173 | evaluate_round 2: strategy sampled 3 clients (out of 3)
DEBUG flwr 2023-09-18 17:41:31,070 | server.py:187 | evaluate_round 2 received 3 results and 0 failures
DEBUG flwr 2023-09-18 17:41:31,071 | server.py:223 | fit_round 3: strategy sampled 3 clients (out of 3)
DEBUG flwr 2023-09-18 17:41:31,095 | server.py:237 | fit_round 3 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:41:31,109 | server.py:125 | fit progress: (3, 0.40491267511680384, {'accuracy': 0.8066666666666666}, 0.2894255999999995)
DEBUG flwr 2023-09-18 17:41:31,110 | server.py:173 | evaluate_round 3: strategy sampled 3 clients (out of 3)
DEBUG flwr 2023-09-18 17:41:31,131 | server.py:187 | evaluate_round 3 received 3 results and 0 failures
DEBUG flwr 2023-09-18 17:41:31,131 | server.py:223 | fit_round 4: strategy sampled 4 clients (out of 4)
DEBUG flwr 2023-09-18 17:41:31,168 | server.py:237 | fit_round 4 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:41:31,179 | server.py:125 | fit progress: (4, 0.4084813456766361, {'accuracy': 0.81}, 0.3589915999999995)
DEBUG flwr 2023-09-18 17:41:31,179 | server.py:173 | evaluate_round 4: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 17:41:31,204 | server.py:187 | evaluate_round 4 received 5 results and 0 failures
DEBUG flwr 2023-09-18 17:41:31,204 | server.py:223 | fit_round 5: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:41:31,236 | server.py:237 | fit_round 5 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:41:31,248 | server.py:125 | fit progress: (5, 0.41013210309633463, {'accuracy': 0.81}, 0.4276208999999991)
DEBUG flwr 2023-09-18 17:41:31,248 | server.py:173 | evaluate_round 5: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:41:31,273 | server.py:187 | evaluate_round 5 received 6 results and 0 failures
DEBUG flwr 2023-09-18 17:41:31,274 | server.py:223 | fit_round 6: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:41:31,309 | server.py:237 | fit_round 6 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:41:31,322 | server.py:125 | fit progress: (6, 0.41013070739986435, {'accuracy': 0.81}, 0.5020378000000001)
DEBUG flwr 2023-09-18 17:41:31,322 | server.py:173 | evaluate_round 6: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:41:31,349 | server.py:187 | evaluate_round 6 received 6 results and 0 failures
DEBUG flwr 2023-09-18 17:41:31,349 | server.py:223 | fit_round 7: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:41:31,387 | server.py:237 | fit_round 7 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:41:31,401 | server.py:125 | fit progress: (7, 0.41013069865829477, {'accuracy': 0.81}, 0.5813123000000004)
DEBUG flwr 2023-09-18 17:41:31,402 | server.py:173 | evaluate_round 7: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:41:31,435 | server.py:187 | evaluate_round 7 received 6 results and 0 failures
DEBUG flwr 2023-09-18 17:41:31,436 | server.py:223 | fit_round 8: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:41:31,472 | server.py:237 | fit_round 8 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:41:31,486 | server.py:125 | fit progress: (8, 0.41013069862808826, {'accuracy': 0.81}, 0.6660415000000004)
DEBUG flwr 2023-09-18 17:41:31,487 | server.py:173 | evaluate_round 8: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:41:31,513 | server.py:187 | evaluate_round 8 received 6 results and 0 failures
DEBUG flwr 2023-09-18 17:41:31,513 | server.py:223 | fit_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:41:31,551 | server.py:237 | fit_round 9 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:41:31,563 | server.py:125 | fit progress: (9, 0.4101306986280269, {'accuracy': 0.81}, 0.7431920999999999)
DEBUG flwr 2023-09-18 17:41:31,563 | server.py:173 | evaluate_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:41:31,590 | server.py:187 | evaluate_round 9 received 6 results and 0 failures
DEBUG flwr 2023-09-18 17:41:31,590 | server.py:223 | fit_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:41:31,629 | server.py:237 | fit_round 10 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:41:31,642 | server.py:125 | fit progress: (10, 0.41013069862802687, {'accuracy': 0.81}, 0.8223275000000001)
DEBUG flwr 2023-09-18 17:41:31,643 | server.py:173 | evaluate_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:41:31,674 | server.py:187 | evaluate_round 10 received 6 results and 0 failures
INFO flwr 2023-09-18 17:41:31,674 | server.py:147 | FL finished in 0.8539812999999992
INFO flwr 2023-09-18 17:41:31,675 | app.py:218 | app_fit: losses_distributed [(1, 0.41476794818173285), (2, 0.4147230205328568), (3, 0.41472413876782294), (4, 0.41185118565483697), (5, 0.40941841219152725), (6, 0.4094184249639511), (7, 0.4094184249639511), (8, 0.4094184249639511), (9, 0.4094184249639511), (10, 0.4094184249639511)]
INFO flwr 2023-09-18 17:41:31,675 | app.py:219 | app_fit: metrics_distributed_fit {}
INFO flwr 2023-09-18 17:41:31,675 | app.py:220 | app_fit: metrics_distributed {}
INFO flwr 2023-09-18 17:41:31,675 | app.py:221 | app_fit: losses_centralized [(0, 0.6931471805599453), (1, 0.4043851898592003), (2, 0.40490823562580686), (3, 0.40491267511680384), (4, 0.4084813456766361), (5, 0.41013210309633463), (6, 0.41013070739986435), (7, 0.41013069865829477), (8, 0.41013069862808826), (9, 0.4101306986280269), (10, 0.41013069862802687)]
INFO flwr 2023-09-18 17:41:31,675 | app.py:222 | app_fit: metrics_centralized {'accuracy': [(0, 0.46), (1, 0.8066666666666666), (2, 0.8066666666666666), (3, 0.8066666666666666), (4, 0.81), (5, 0.81), (6, 0.81), (7, 0.81), (8, 0.81), (9, 0.81), (10, 0.81)]}
>>>>>>>>>>>>> setting initial parameters
[[0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0.]]
>>>>>>>>>>>>> evaluating on server-side
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.98483899,  0.99983765,  0.71273568,  0.46293981,  0.18966543,
         0.08786642,  0.01543199, -0.04161428]]), array([0.08489786])]
[array([[-1.98483899,  0.99983765,  0.71273568,  0.46293981,  0.18966543,
         0.08786642,  0.01543199, -0.04161428]]), array([0.08489786])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.96078494,  1.01397584,  0.72125967,  0.45417681,  0.17774835,
         0.07617043,  0.03384813, -0.04876843]]), array([0.04860261])]
[array([[-1.96078494,  1.01397584,  0.72125967,  0.45417681,  0.17774835,
         0.07617043,  0.03384813, -0.04876843]]), array([0.04860261])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.96045138,  1.01380176,  0.72108734,  0.45413275,  0.17771861,
         0.07614749,  0.03378604, -0.04882921]]), array([0.0487367])]
[array([[-1.96045138,  1.01380176,  0.72108734,  0.45413275,  0.17771861,
         0.07614749,  0.03378604, -0.04882921]]), array([0.0487367])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.91229883,  0.96622081,  0.78022721,  0.37762582,  0.17410835,
         0.07055901, -0.01566226, -0.03741224]]), array([0.1358597])]
[array([[-1.91229883,  0.96622081,  0.78022721,  0.37762582,  0.17410835,
         0.07055901, -0.01566226, -0.03741224]]), array([0.1358597])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.96576327,  1.00060977,  0.83502706,  0.37455868,  0.15904389,
         0.0775111 , -0.02725578, -0.05174605]]), array([0.16891901])]
[array([[-1.96576327,  1.00060977,  0.83502706,  0.37455868,  0.15904389,
         0.0775111 , -0.02725578, -0.05174605]]), array([0.16891901])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.96570291,  1.00067181,  0.83499924,  0.37471523,  0.15903256,
         0.07751956, -0.02729247, -0.05173999]]), array([0.16867228])]
[array([[-1.96570291,  1.00067181,  0.83499924,  0.37471523,  0.15903256,
         0.07751956, -0.02729247, -0.05173999]]), array([0.16867228])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.96570296,  1.00067173,  0.83499909,  0.37471545,  0.15903257,
         0.07751981, -0.02729256, -0.05174039]]), array([0.16867244])]
[array([[-1.96570296,  1.00067173,  0.83499909,  0.37471545,  0.15903257,
         0.07751981, -0.02729256, -0.05174039]]), array([0.16867244])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.96570296,  1.00067173,  0.83499909,  0.37471545,  0.15903257,
         0.07751981, -0.02729256, -0.05174039]]), array([0.16867245])]
[array([[-1.96570296,  1.00067173,  0.83499909,  0.37471545,  0.15903257,
         0.07751981, -0.02729256, -0.05174039]]), array([0.16867245])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.96570296,  1.00067173,  0.83499909,  0.37471545,  0.15903257,
         0.07751981, -0.02729256, -0.05174039]]), array([0.16867245])]
[array([[-1.96570296,  1.00067173,  0.83499909,  0.37471545,  0.15903257,
         0.07751981, -0.02729256, -0.05174039]]), array([0.16867245])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.96570296,  1.00067173,  0.83499909,  0.37471545,  0.15903257,
         0.07751981, -0.02729256, -0.05174039]]), array([0.16867245])]
[array([[-1.96570296,  1.00067173,  0.83499909,  0.37471545,  0.15903257,
         0.07751981, -0.02729256, -0.05174039]]), array([0.16867245])]
>>>>>>>>>>>>>run time:
0.53125s
